### PR TITLE
Fixes #697 Exception while editing IronPython code

### LINKED ISSUE #697: Exception while editing IronPython code
VS2013 Desktop Express, IronPython 2.7.5, PTVS 2.2.30718.00

Occasionally while editing IronPython code I get a modal MessageBox telling me that 
"An exception has been encountered. This may be caused by an extension."

The relevant entry in the VS activity log starts

System.InvalidCastException: Unable to cast object of type 'IronPython.Runtime.Types.BuiltinFunction' to type 'IronPython.Runtime.Types.PythonType'. at
Microsoft.IronPythonTools.Interpreter.RemoteInterpreter.<>c__DisplayClass64_0.<GetPythonTypeMemberType>b__0() at Microsoft.IronPythonTools.Interpreter.RemoteInterpreter.CallAndHandle[T](Func`1 action, T defaultValue) at
Microsoft.IronPythonTools.Interpreter.RemoteInterpreter.GetPythonTypeMemberType(ObjectIdentityHandle pythonType) at
Microsoft.IronPythonTools.Interpreter.RemoteInterpreterProxy.GetPythonTypeMemberType(ObjectIdentityHandle value) at
Microsoft.IronPythonTools.Interpreter.RemoteInterpreterProxy.GetPythonTypeMemberType(ObjectIdentityHandle value) at
Microsoft.IronPythonTools.Interpreter.IronPythonType.get_MemberType() at
Microsoft.PythonTools.Analysis.Values.BuiltinClassInfo.get_MemberType() at
Microsoft.PythonTools.Analysis.Values.SpecializedNamespace.get_MemberType() at
Microsoft.PythonTools.Analysis.MemberResult.GetMemberType() at
Microsoft.PythonTools.Analysis.MemberResult.get_MemberType() at
Microsoft.PythonTools.Intellisense.CompletionAnalysis.PythonCompletion(IGlyphService service, MemberResult memberResult) at
...  

Unfortunately I can't see any way of attaching the activity log here.

### PR DESCRIPTION
**Pull request by zooba**
Fixes #697 Exception while editing IronPython code
Ensures generic types are wrapped and unwrapped correctly.

### CODE REVIEW

**Comment by zooba** [comment]
Ping @DinoV

File: Python/Product/IronPython/Interpreter/RemoteInterpreter.cs [hunk 1]
```diff
@@ -1055,9 +1055,10 @@ private class CannotUnwrapHandleException : Exception {
                 if (genType != null) {
                     Type[] genTypes = new Type[types.Length];
                     for (int i = 0; i < types.Length; i++) {
-                        genTypes[i] = (Type)Unwrap(types[i]);
+                        var o = Unwrap(types[i]);
+                        genTypes[i] = o as Type ?? (PythonType)o;
```
Comment (DinoV) [inline Python/Product/IronPython/Interpreter/RemoteInterpreter.cs]: You're putting a PythonType into a Type array?

Comment (zooba) [inline Python/Product/IronPython/Interpreter/RemoteInterpreter.cs]: It has an implicit cast to Type that isn't handled by the `as` cast (C# won't do object->PythonType->Type in one step).

**Comment by DinoV** [comment]
LGTM

**Pull request merged**